### DISCUSSION TITLE
Retrieving Data from Table via GET-Request [Python]

### DISCUSSION THREAD

**Discussion by Den4ik1992** [Budibase/budibase#10657] [Help] [answered]
Hello everyone, I want to retrieve the entire table from my database using a web request. I am using this Python script:

`import requests
url = "https://budibase.app/api/public/v1/tables/tableid"
headers = {
    "accept": "application/json",
    "x-budibase-app-id": "myappid",
    "x-budibase-api-key": myapikey",
}
response = requests.get(url, headers=headers)
print(response.text)
`

Everything seems to be working fine, but I only get the table "headers" as JSON data and not the actual data in the table. I'm able to retrieve the name and type of the columns, but I'm not getting any row data.

What am I doing wrong ?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Den4ik1992,

This looks like you are only querying the table. In this case you want to query the rows within that table. Here is the relevant public api documentation which you require. https://docs.budibase.com/reference/post_tables-tableid-rows For this you will want the app id and the table id as well as your api key to configure this properly. app_id and tableid can all be found in your url where you view your table. 
<img width="606" alt="Screenshot 2023-05-22 at 11 37 58" src="https://github.com/Budibase/budibase/assets/126772285/6552a301-b868-4959-815f-605516f4fcbe">


**Key note**, if you have more than 1000s rows you will have you use the bookmarking system to query additional results in batches of 1000.

Hopefully this helps solve your issue!
